tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/PublicAssetSetMethodContract.scala

```diff
@@ -1963,4 +1972,106 @@ trait PublicAssetSetMethodContract {
       .jsonPath()
       .get("methodResponses[0][1].created.4f29.id")
   }
+
+  @Test
+  def createShouldReturnFailWhenPublicAssetQuotaLimitIsExceeded(): Unit = {
+    val content = "Your asset content here".repeat(10).getBytes
+    val uploadResponse: UploadResponse = uploadAsset(content)
+    val uploadResponse2: UploadResponse = uploadAsset(content)
+    val uploadResponse3: UploadResponse = uploadAsset(content)
+
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+         |  "methodCalls": [
+         |    [
+         |      "PublicAsset/set", {
+         |        "accountId": "$ACCOUNT_ID",
+         |        "create": {
+         |          "4f29": {
+         |            "blobId": "${uploadResponse.blobId}"
+         |          }
+         |        }
+         |      }, "0"
+         |    ]
+         |  ]
+         |}""".stripMargin
+
+    `given`()
+      .body(request)
+      .when()
+      .post()
+      .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    val request2: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+         |  "methodCalls": [
+         |    [
+         |      "PublicAsset/set", {
+         |        "accountId": "$ACCOUNT_ID",
+         |        "create": {
+         |          "4f29": {
+         |            "blobId": "${uploadResponse2.blobId}"
+         |          }
+         |        }
+         |      }, "0"
+         |    ]
+         |  ]
+         |}""".stripMargin
+
+    `given`()
+      .body(request2)
+      .when()
+      .post()
+      .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    val request3: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+         |  "methodCalls": [
+         |    [
+         |      "PublicAsset/set", {
+         |        "accountId": "$ACCOUNT_ID",
+         |        "create": {
+         |          "4f29": {
+         |            "blobId": "${uploadResponse3.blobId}"
+         |          }
+         |        }
+         |      }, "0"
+         |    ]
+         |  ]
+         |}""".stripMargin
+
+    val response: String = `given`()
+      .body(request3)
+      .when()
+      .post()
+      .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1].notCreated")
+      .isEqualTo(
+        s"""{
+           |    "4f29": {
+           |        "type": "serverFail",
```

Comment:
ServerFail is definitly not the good error to report and shows the absence of correct error management.